op-challenger/game/fault/trace/vm/executor.go

```diff
@@ -125,3 +115,40 @@ func (e *Executor) DoGenerateProof(ctx context.Context, dir string, begin uint64
 	e.metrics.RecordVmExecutionTime(e.cfg.VmType.String(), time.Since(execStart))
 	return err
 }
+
+func (e *Executor) fillHostCommand(args []string, dataDir string) []string {
+	switch e.serverType {
+	case types.ServerTypeOpProgram:
+		args = append(args,
+			"--",
+			e.cfg.Server, "--server",
+			"--l1", e.cfg.L1,
+			"--l1.beacon", e.cfg.L1Beacon,
+			"--l2", e.cfg.L2,
+			"--datadir", dataDir,
+			"--l1.head", e.inputs.L1Head.Hex(),
+			"--l2.head", e.inputs.L2Head.Hex(),
+			"--l2.outputroot", e.inputs.L2OutputRoot.Hex(),
+			"--l2.claim", e.inputs.L2Claim.Hex(),
+			"--l2.blocknumber", e.inputs.L2BlockNumber.Text(10),
+		)
+		return args
+	case types.ServerTypeKona:
+		args = append(args,
```

Comment:
This would be a good place to use branch by abstraction rather than having to put every implementation we ever need to support into this file with a giant `switch` case.

So rather than passing in an enum for the server type, pass in an implementation of `ServerArguments` interface.  It would have a message like `ServerArgs(dataDir string, vmConfig Config, localInputs utils.LocalGameInputs) []string`.